pkg/resources/resources_test.go

```diff
@@ -114,6 +116,76 @@ func TestCheckResourceStatus(t *testing.T) {
 		assert.Error(t, err)
 		assert.Equal(t, "unsupported resource type", err.Error())
 	})
+
+	t.Run("Should return error when DeploymentProcessing is False", func(t *testing.T) {
+		dep := &appsv1.Deployment{
+			Status: appsv1.DeploymentStatus{
+				ReadyReplicas: 3,
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:    appsv1.DeploymentProgressing,
+						Status:  corev1.ConditionFalse,
+						Reason:  "ProcessDeadlineExceeded",
+						Message: "Deployment exceeded its progress deadline",
+					},
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: int32Ptr(3),
+			},
+		}
+		cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(dep).Build()
+		err := CheckResourceStatus(dep, cl, 2*time.Second)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "Deployment exceeded its progress deadline")
+	})
+
+	t.Run("Should return error for Job with failed pods", func(t *testing.T) {
+		job := &batchv1.Job{
+			Status: batchv1.JobStatus{
+				Failed: 1,
+			},
+		}
+		cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(job).Build()
+		err := CheckResourceStatus(job, cl, 2*time.Second)
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "has failed 1 pods")
+	})
+
+	t.Run("Should return deadline exceeded for Job with only active pods", func(t *testing.T) {
+		job := &batchv1.Job{
+			Status: batchv1.JobStatus{
+				Active: 1,
+			},
+		}
+		cl := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(job).Build()
+		err := CheckResourceStatus(job, cl, 2*time.Second)
+		assert.Error(t, err)
+		assert.Equal(t, err, context.DeadlineExceeded)
+	})
```

Comment:
Does this take 30 minutes to run? Or where do you set timeout limit for this

Reply:
Time limit is 2 second, set it on line 162

Reply:
oh woops overlooked, nice